src/classes/PermissionsHelper.php

```diff
@@ -56,4 +58,24 @@ public function getAssociativeArray(): array
         }
         return $base;
     }
+
+    /**
+     * Builds an array used by extended search
+     */
+    public function getExtendedSearchAssociativeArray(): array
```

Comment:
I really don't like this new function, and I believe there is a way to avoid having this. I don't know what it is, but I'm sure there is. Might be easier said than done, but it's worth trying.

Reply:
What about this function do you not like?
Is it the fact it is in PermissionsHelper?
This can be moved into VisibilityFieldHelper.

I think the plain english terms should be there but maybe they can be dropped and this function would not be necessary.

Reply:
I'm thinking extending the enum should be better here, with `toKey()` for example. And I don't like the code duplication with the useronly if.

Maybe it would be possible to remove this visibility arr that is passed around. What if VisibilityFieldHelper would get it by itself from the enum? I don't have this part of the code in mind though, maybe there is another reason that this must be in the VisitorParameters construct.

This new permission system is probably calling for a more profound rework of the extended search, now that teamgroups are extracted from the previous "visibilityArr".

Reply:
> This new permission system is probably calling for a more profound rework of the extended search, now that teamgroups are extracted from the previous "visibilityArr".

Yes, this needs to be addressed in a better way. Do you want to keep the ability to search for entities based on their teamgroups visibility and maybe extend it to the user level?

Based on this decision it might make sense to add a flag to specify what is searched e.g `visibility:tg:"Team Name"` or `visibility:u:"User Name"`.

> Maybe it would be possible to remove this visibility arr that is passed around. What if VisibilityFieldHelper would get it by itself from the enum? I don't have this part of the code in mind though, maybe there is another reason that this must be in the VisitorParameters construct.

Yes, now the enum can be called from within the VisibilityFieldHelper because it is not depended on `Users` anymore.
However, if teamgroups shall be included this information must be passed via the VisitorParameters.

Reply:
I'll decide on that later.